compiler/rustc_lint/src/types.rs

```diff
@@ -727,7 +728,48 @@ declare_lint! {
     "proper use of libc types in foreign item definitions"
 }
 
-declare_lint_pass!(ImproperCTypesDefinitions => [IMPROPER_CTYPES_DEFINITIONS]);
+declare_lint! {
+    /// In it's platform C ABI, the AIX target follows the power alignment rule
+    /// for structs. This rule applies the natural alignment of the first struct
+    /// field if it is a floating-point type that is greater than 4-bytes, or
+    /// is an aggregate whose recursively "first" field is a floating-point
+    /// type greater than 4-bytes.
+    /// The alignment for subsequent fields of the associated structs use an
+    /// alignment value where the floating-point type is aligned on a 4-byte
```

Comment:
an important question: how would the XLC compiler handle this case?

```c
#pragma align(power)
typedef struct Lol {
    float criminal_field;
    uint16_t filler;
    _Float16 half_as_bad;
} Lmao;
```

Does the `_Float16` get aligned to 4? Does the compiler even support a half type of some kind? (please substitute whatever it does support, if so).

Reply:
Using XLC, it appears that half precision floating point was never supported in C on the AIX platform. Thus, the ABI is currently unknown.

Reply:
I see.